doc/05_Objects/05_External_System_Interaction.md

```diff
@@ -94,6 +94,7 @@ If you're using / creating very much objects you should call the Pimcore garbage
 ```
 
 **WARNING:** This will flush the entire internal registry!
+**NOTICE:** This helper only works in `prod` environments!
```

Comment:
could you please elaborate your point? 

This calls cleanup on long running helper which in turn closes doctrine connections, cleans up monolog handlers, runtime cache and calls php garbage collector.

Reply:
@dvesh3 we've figured out that the helper call is working and freeing the memory only if the APP_ENV is set to prod, otherwise it doesn't cleanup the memory.
Tried with 3 different custom commands, parsing and creating thousand of records. Once set to prod, the memory is cleared correctly, otherwise not.
We've already created an issue some days ago: https://github.com/pimcore/pimcore/issues/15866

Reply:
I don't see that there are distinctions based on the `APP_ENV` set as `prod`, could it be the opposite instead? that in `dev` mode the cache might be disabled and behaving differently? https://github.com/pimcore/pimcore/blob/e75347322c7385d477feb7152e62eaebd8127b48/doc/19_Development_Tools_and_Details/13_Debugging.md?plain=1#L18

Reply:
@kingjia90 i agree with you, I haven't done some reverse engineering to inspect the behaviour but is for sure related to the caching engine. I suppose it would be helpful to know that in DEV mode the helper has a different behaviour than the PROD mode, do you agree?

Reply:
Yes, I agree, if it's behaving differently, it should be documented, but i am not so much into the topic yet to guess what's going on. 
We may need to first rule out if that was intended or is just undocumented or if is a potential bug. Looks more like a bug to me 🙈